components/codec/src/byte.rs

```diff
@@ -98,6 +98,56 @@ impl MemComparableByteCodec {
         }
     }
 
+    /// Encodes all bytes in the `src` in ascending memory-comparable format in place.
+    /// If the capacity of `src` is less than the encoded length, `src` will `reserve` the
+    /// minimum sufficient capacity.
+    ///
+    /// Returns the number of bytes encoded.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the encoded length overflows usize.
+    pub fn encode_all_in_place(src: &mut Vec<u8>) -> usize {
```

Comment:
Prefer to accept `&mut [u8]` instead of a Vec, so that other kind of write targets can also be supported (for example, a pre-allocated buffer). You can check whether there is enough space in this function, to leave the role of extending the Vec to the caller.

Reply:
Got it.